### PR TITLE
Preserve errors returned by diskInfo to detect disk errors

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Preserve errors returned by diskInfo to detect disk errors

## Motivation and Context
This PR basically reverts #9720 and re-implements it differently

## How to test this PR?
Same test as mentioned in #9720 to detect disk unmounts

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed
- [ ] Functional tests needed (If yes, add [mint](https://github.com/minio/mint) PR # here: )

### CODE REVIEW

File: cmd/storage-rest-client.go [hunk 1]
```diff
@@ -177,19 +177,11 @@ func (client *storageRESTClient) CrawlAndGetDataUsage(ctx context.Context, cache
 }
 
 func (client *storageRESTClient) GetDiskID() (string, error) {
-	respBody, err := client.call(storageRESTMethodGetDiskID, nil, nil, -1)
```
Comment (vadmeste) [inline cmd/storage-rest-client.go:180]: Can you also remove in server's side ? DiskIDHandler()

**Review by vadmeste — COMMENTED** [review]
One commet, LGTM otherwise

**Review by vadmeste — APPROVED** [review]
LGTM

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**